src/Backends/PrefersColorSchemeSettings.vala

```diff
@@ -0,0 +1,120 @@
+/*
+* Copyright 2020 elementary, Inc. (https://elementary.io)
+*
+* This program is free software; you can redistribute it and/or
+* modify it under the terms of the GNU General Public
+* License as published by the Free Software Foundation; either
+* version 3 of the License, or (at your option) any later version.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+* General Public License for more details.
+*
+* You should have received a copy of the GNU General Public
+* License along with this program; if not, write to the
+* Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+* Boston, MA 02110-1301 USA
+*
+* Authored by: Marius Meisenzahl <mariusmeisenzahl@gmail.com>
+*/
+
+public class SettingsDaemon.Backends.PrefersColorSchemeSettings : GLib.Object {
+    public unowned PantheonShell.Pantheon.AccountsService accounts_service { get; construct; }
+
+    private GLib.Settings color_settings;
+    private double pos_lat = -1.0;
+    private double pos_long = -1.0;
+
+    public PrefersColorSchemeSettings (PantheonShell.Pantheon.AccountsService accounts_service) {
+        Object (accounts_service: accounts_service);
+    }
+
+    construct {
+        color_settings = new GLib.Settings ("io.elementary.settings-daemon.prefers-color-scheme");
+
+        get_location.begin ();
```

Comment:
I think it's probably best not to get the location and set up a 1s `TimeoutSource` if this option is disabled.

We should only get the location if we need it, and only set up the `TimeoutSource` if this feature is not disabled.

Then this can listen to the settings key and start/stop the `TimeoutSource` and get the location as needed.

Reply:
I tried to optimize the logic to query the location and start/stop `TimeoutSource`.

I'm quite satisfied with the logic for the location query. But I would be very happy about feedback for starting and stopping `TimeoutSource` 🙈